uber/site_sections/registration.py

```diff
@@ -801,6 +801,54 @@ def review(self, session):
                                     .order_by(Attendee.full_name).all()}
 
     @site_mappable
+    def promo(self, session, message='', **params):
+        pc = session.promo_code(params)
+        now = datetime.now(UTC)
+        today = "%d-%02d-%02d" % (now.date().year, now.date().month, now.date().day)
+        max_day = "%d-%02d-%02d" % (c.EPOCH.date().year, c.EPOCH.date().month, c.EPOCH.date().day)
+        if 'price' in params:
+            if pc.expiration_date == '':
```

Comment:
This could be cleaned up a bit by just saying `if `price` in params and pc.expiration_date`. Python evaluates empty strings to false so you (usually) don't need to explicitly check for them.